src/app/reporting/Engine.cpp

```diff
@@ -373,14 +420,14 @@ CHIP_ERROR Engine::BuildSingleReportDataEventReports(ReportDataMessage::Builder
     ChipLogDetail(DataManagement, "Fetched %u events", static_cast<unsigned int>(eventCount));
 
 exit:
+    hasEncodedData = hasEncodedData || (eventCount != 0);
     if (apHasEncodedData != nullptr)
     {
-        *apHasEncodedData = !(eventCount == 0 || eventClean);
+        *apHasEncodedData = hasEncodedData;
```

Comment:
I'd think the `hasEncodedData = hasEncodedData || ... ` is kinda redundant and confusing. You can just have `*apHasEncodedData = hasEncodedData || (eventCount != 0);`.

I also think renaming `hasEncodedData` to `hasEncodedStatus` makes more sense, since it is specifically tracking just encoding status, while eventCount is tracking encoding 'data' right?

Reply:
done, thanks